src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -77,11 +81,22 @@ const char SDK_WWW_NATPF_RULE_NAME[] = "guestwww";
 const char SDK_WWW_NATPF_RULE_TEMPLATE[] = "guestwww,tcp,127.0.0.1,%1,,9292";
 const char EMULATOR_SSH_NATPF_RULE_NAME[] = "guestssh";
 const char EMULATOR_SSH_NATPF_RULE_TEMPLATE[] = "guestssh,tcp,127.0.0.1,%1,,22";
+const char MODIFYMEDIUM[] = "modifymedium";
+const char RESIZE[] = "--resize";
+const char MEMORY[] = "--memory";
+const char CPUS[] = "--cpus";
+const char SHOWMEDIUMINFO[] = "showmediuminfo";
+const char METRICS[] = "metrics";
+const char COLLECT[] = "collect";
+const char TOTAL_RAM[] = "RAM/Usage/Total";
 
 namespace Mer {
 namespace Internal {
 
-static VirtualMachineInfo virtualMachineInfoFromOutput(const QString &output);
+static VirtualMachineInfo virtualMachineInfoFromOutput(const QString &output, bool fillVdiInfo);
+static void vdiInfoFromOutput(const QString &output, VirtualMachineInfo &virtualMachineInfo);
+static void fetchVdiInfo(VirtualMachineInfo &virtualMachineInfo);
```

Comment:
virtualMachineInfo should be passed by pointer. Please also swap the two declarations: fetchVdiInfo first, vdiInfoFromOutput second.